osal/linux/osal_common.c

```diff
@@ -43,17 +43,18 @@ static void osal_alarm_fired();
  *
  * @brief     create new task/thread. 
  * input parameters
- *
+ *  @param[in] thread address to store thread ID
  *  @param[in] name name of the task
  *  @param[in] priotiy of task 
  *  @param[in] stacksize size of the thread stack.
  *  @param[in] entry the task is created executing entry 
  *  @param[in] arg entry routine arguments 
  *
  * output parameters
- * @return    osal_task_t ID of the created task/thread
+ * @return    osal_basetype_t on success return 0 otherwise error value
  *****************************************************************************/
-osal_task_t osal_task_create (
+osal_basetype_t osal_task_create(
+   osal_task_t * thread,
```

Comment:
Any specific reason for passing *thread as parameter to osal_task_create()

Reply:
Yep, my reason are:
- to avoid returning a deep copy of a structure
- standardise the API return value

Reply:
Thanks for clarifying.